db/post_migrate/20201111005205_move_data_to_event_dates.rb

```diff
@@ -1,41 +1,103 @@
 # frozen_string_literal: true
 
 class MoveDataToEventDates < ActiveRecord::Migration[6.0]
+
+  VALID_OPTIONS = [
+    :start,
+    :end,
+    :status,
+    :"allowed-groups",
+    :url,
+    :name,
+    :reminders,
+    :recurrence
+  ]
+
+  def extract_events(post)
+    cooked = PrettyText.cook(post.raw, topic_id: post.topic_id, user_id: post.user_id)
+    valid_options = VALID_OPTIONS.map { |o| "data-#{o}" }
+
+    valid_custom_fields = []
+    SiteSetting.discourse_post_event_allowed_custom_fields.split('|').each do |setting|
```

Comment:
small one, but this relies on the site setting which may be renamed... instead just select form the table